state/it/src/test/java/com/expediagroup/streamplatform/streamregistry/state/it/AgentIT.java

```diff
@@ -0,0 +1,276 @@
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.state.it;
+
+import static com.expediagroup.streamplatform.streamregistry.state.model.event.Event.specificationDeletion;
+import static java.util.UUID.randomUUID;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.aMapWithSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.hamcrest.collection.IsIterableContainingInOrder.contains;
+import static org.hamcrest.collection.IsMapContaining.hasEntry;
+import static org.hamcrest.core.IsIterableContaining.hasItem;
+import static org.hamcrest.core.IsIterableContaining.hasItems;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.stream.Collectors;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.SneakyThrows;
+import lombok.val;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.awaitility.Awaitility;
+import org.awaitility.core.ConditionFactory;
+import org.junit.Rule;
+import org.junit.Test;
+import org.testcontainers.containers.KafkaContainer;
+
+import com.expediagroup.streamplatform.streamregistry.state.AgentData;
+import com.expediagroup.streamplatform.streamregistry.state.DefaultEntityView;
+import com.expediagroup.streamplatform.streamregistry.state.EntityView;
+import com.expediagroup.streamplatform.streamregistry.state.EntityViewListener;
+import com.expediagroup.streamplatform.streamregistry.state.EventSender;
+import com.expediagroup.streamplatform.streamregistry.state.kafka.KafkaEventReceiver;
+import com.expediagroup.streamplatform.streamregistry.state.kafka.KafkaEventSender;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.DomainKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.Key;
+import com.expediagroup.streamplatform.streamregistry.state.model.event.Event;
+import com.expediagroup.streamplatform.streamregistry.state.model.specification.DefaultSpecification;
+import com.expediagroup.streamplatform.streamregistry.state.model.specification.Specification;
+
+public class AgentIT {
+
+  private final String schemaRegistryUrl = "mock://schemas";
+
+  private final ConditionFactory await = Awaitility.await()
+    .atMost(10, SECONDS)
+    .pollDelay(1, SECONDS);
+
+  @Rule
+  public KafkaContainer kafka = new KafkaContainer();
+
+  @Test
+  public void testBootstrapping() {
+    val topicName = topicName();
+    val kafkaEventSender = kafkaEventSender(topicName);
+    val entityView = new DefaultEntityView(kafkaEventReceiver(topicName, "groupId"));
+    val dummyAgent = new StoringEntityViewListener();
```

Comment:
I think you could add this to a BeforeEach? These lines occur in every test right?